### ISSUE TITLE
Checkboxgroup is not working as expected in the SmartForm

### ISSUE THREAD

**Issue by eddielisands** [VulcanJS/Vulcan#1998]
I tried with the "example-forum" in the latest "Vulcan-Starter" repo, I could add the new category, but could not remove the checked category from the "Edit Post" after submitting the form.
<img width="922" alt="screen shot 2018-06-13 at 10 40 31" src="https://user-images.githubusercontent.com/6302149/41339862-3f33348c-6ef6-11e8-965d-e3e32cbbdebc.png">

**Comment by SachaG** [comment]
OK, I see the issue now. It's a bit tricky to fix, I need to think about it…

**Comment by SachaG** [comment]
Yes it was because of the rewrite. You can see how I fixed it in the linked commit.

**Comment by eddielisands** [comment]
The label of the checkboxgroup is disappeared, and the 'null' value causes the validation error if the data type is 'String'.

**Comment by Apollinaire** [comment]
Yes, I can confirm that it's broken on the current devel. It works on the master branch though (v1.11.2). I don't know what could have broken this, I did not read the latest commits
And the label is missing too.

**Comment by eddielisands** [comment]
Add <label>{inputProperties.label}</label> to the beginning of the <div> could add the label back.
The default value for the 'checkboxgroup' is [""], and it will update to [undefined, "value"] if you toggle the second checkbox. The "undefined" value breaks the graphQL validation when submitting the smart form.

**Comment by SachaG** [comment]
I fixed the missing label. The null value thing is trickier, let me think about it…

**Comment by SachaG** [comment]
So the root issue is deciding whether the array containing the checkbox values can be "sparse" or not, i.e. `['foo', null, 'bar']` vs `['foo', 'bar']`.

With nested forms (like in the address example) it's important to preserve empty array items because we rely on paths to update other items – so if we delete the first item, we don't want the `foo.0` path to start pointing to the previously second –and now first– item.

For checkboxes though it makes less sense, so maybe we shouldn't be using that same system at all and just handle the entire checkbox array's state inside the checkbox group component instead.

**Comment by SachaG** [comment]
That should hopefully fix it. It was actually a lot simpler than I thought (assuming it actually works now!).

**Comment by SachaG** [comment]
Actually this is still not fixed because when going from `['a', 'b', 'c']` to `['a', 'c']` the old a new values get merged meaning you still end up with `['a', 'b', 'c']`…
